docs/blog/2022-08-12-observing-randomness.md

```diff
@@ -0,0 +1,140 @@
+
+Especially during ceremonies, it is important to see changes in node state as soon as possible (< 1 minute) after they happen.
+
+## Design decisions
+
+The above requirements led us to the following design:
+
+**Each node exposes its own metrics in Prometheus format**
+
+Prometheus is an industry-standard format, and each node exposing its own metrics allows anyone to monitor their own node directly.
+
+**Additionally, each node is able to retrieve any other connected node’s metrics and expose them in Prometheus format**
+
+This enables not only Protocol Labs but any LoE member to create their own dashboards which show the state of the whole network.
+
+**The transport of metrics between nodes is done using the same mechanism that drand uses for inter-node communication when generating randomness (GRPC)**
+
+With this we allow metrics to be sent between nodes, while not requiring opening any additional network ports.
+
+**For the Protocol Labs dashboard, gather the metrics using Telegraf and export them to InfluxDB, which we can access via Grafana Cloud**
+
+This enables us to gather metrics from the whole swarm into a central location and create an easy-to-use dashboard. The scraping interval is set to one minute, and it can be lowered during ceremonies in order to get faster network state updates.
+
+## Metrics design
+
+To start with, we had to decide what we needed to monitor on each node. We came up with the following minimal information requirements for running a smooth ceremony:
+
+- Version of drand that the node is running, in order to ensure that all nodes are running the same/correct version.
+- Whether nodes have restarted their daemon ahead of the ceremony, to ensure that the certificate is still valid and to clear the process’ memory and storage.
+- Whether nodes have connectivity to all other nodes.
+- When and if the node started the ceremony process.
+- What is the current state of the node related to the ceremony.
+
+We also came up with the idea of using the dashboard for ongoing monitoring of the network (not only during ceremonies), for which we added the following for each beacon in each node:
+
+- When was the last randomness generated.
+- Which is the last round generated by the node.
+- What is the time difference between this node and other nodes (including network round-trip times).
+
+The drand code was already instrumented to provide some of the needed metrics via a Prometheus HTTP endpoint, while others needed to be added.
+
+## Inter-node communications
+
+As mentioned above, one of the requirements was to be able to view the state of the whole drand network without requiring node operators to open any additional holes in their firewall, nor to connect to any hosts other than those included in the drand network itself. Therefore, it was necessary to enable the possibility of fetching a remote node’s metrics via the same method used to coordinate randomness generation: that is, through GRPC.
```

Comment:
```suggestion
As mentioned above, one of the requirements was to be able to view the state of the whole drand network without requiring node operators to open any additional holes in their firewall, nor to connect to any host other than those included in the drand network itself. Therefore, it was necessary to enable the possibility of fetching a remote node’s metrics via the same method used to coordinate randomness generation: that is, through GRPC.
```

Reply:
Same as with `ports` vs `port` elsewhere, to me the plural feels "more correct".